lib/src/test/resources/compileContracts.sh

```diff
@@ -1,47 +1,113 @@
 #!/bin/bash
 
-pushd "$(dirname "$0")" || exit
-mkdir -p "compiled_v0"
+V1_COMPILER_BUILD_PATH_VAR="V1_COMPILER_BUILD_PATH"
+V2_COMPILER_BUILD_PATH_VAR="V2_COMPILER_BUILD_PATH"
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+
+V0_CONTRACT_PATH="src_v0"
+V1_CONTRACT_PATH="src_v1"
+V2_CONTRACT_PATH="src_v2"
+
+V0_ARTIFACT_PATH="compiled_v0"
+V1_ARTIFACT_PATH="compiled_v1"
+V2_ARTIFACT_PATH="compiled_v2"
+
+V1_COMPILER_VERSION="1.1.1"
+V2_COMPILER_VERSION="2.2.0"
+
+V1_COMPILER_BIN_PATH="compilers/v1"
+V2_COMPILER_BIN_PATH="compilers/v2"
+
+fetch_compilers() {
+  local VERSION=$1
+  local OUT_DIR=$2
+
+  local VERSION_SHORT=${VERSION:0:1}
+  [ "$VERSION_SHORT" == "1" ] && local COMPILER_BUILD_PATH_VAR="$V1_COMPILER_BUILD_PATH_VAR"
+  [ "$VERSION_SHORT" == "2" ] && local COMPILER_BUILD_PATH_VAR="$V2_COMPILER_BUILD_PATH_VAR"
+  local COMPILER_BUILD_PATH="${!COMPILER_BUILD_PATH_VAR}"
+
+  echo "Fetching compiler v$VERSION binaries..."
+
+  mkdir -p "$OUT_DIR"
+
+  OS=$(uname)
+  ARCH=$(uname -m)
+
+  if [ "$OS" == "Linux" -a "$ARCH" == "x86_64" ]; then
+    echo "Source: https://github.com/starkware-libs/cairo/releases/"
+    curl -LsSf "https://github.com/starkware-libs/cairo/releases/download/v${VERSION}/release-x86_64-unknown-linux-musl.tar.gz" | tar -xz -C "$OUT_DIR" || exit 1
+  elif [ "$OS" == "Darwin" -a "$ARCH" == "arm64" ]; then
+    echo "Source: https://github.com/starkware-libs/cairo/releases/"
+    curl -LsSf "https://github.com/starkware-libs/cairo/releases/download/v${VERSION}/release-aarch64-apple-darwin.tar" | tar -x -C "$OUT_DIR" || exit 1
+  else
+      if [ -n "${COMPILER_BUILD_PATH}" ] && [ -d "${COMPILER_BUILD_PATH}" ]; then
+        echo "Source: $COMPILER_BUILD_PATH..."
+
+        rm -r "$(dirname "$0")/$OUT_DIR/cairo/bin/" || true
+        rm -r "$(dirname "$0")/$OUT_DIR/cairo/corelib/" || true
+
+        mkdir -p "$(dirname "$0")/$OUT_DIR/cairo/bin/"
+        mkdir -p "$(dirname "$0")/$OUT_DIR/cairo/corelib/"
+
+        echo "Copying binaries..."
+        rsync -aW "${COMPILER_BUILD_PATH}/" "$(dirname "$0")/$OUT_DIR/cairo/bin/" || exit 1
+        echo "Copying corelib..."
+        rsync -aW "$REPO_ROOT/cairo$VERSION_SHORT/corelib/" "$(dirname "$0")/$OUT_DIR/cairo/corelib/" || exit 1
+      else
+        echo "Your OS or architecture ($ARCH-$OS) is not supported directly."
+        echo "To proceed, please specify a valid path to a built v$VERSION compiler for ($ARCH-$OS) in the $COMPILER_BUILD_PATH_VAR environment variable."
```

Comment:
not sure if I follow, but from what I see you have to actually export a different var (eg `V1_COMPILER_BUILD_PATH_VAR` or `V2_COMPILER_BUILD_PATH_VAR`)? or am I missing something?

Reply:
Are you referring to `$COMPILER_BUILD_PATH_VAR` in echo?
It is defined above and takes value of `V1_COMPILER_BUILD_PATH_VAR` or `V2_COMPILER_BUILD_PATH_VAR` depending on current version.
```
  [ "$VERSION_SHORT" == "1" ] && local COMPILER_BUILD_PATH_VAR="$V1_COMPILER_BUILD_PATH_VAR"
  [ "$VERSION_SHORT" == "2" ] && local COMPILER_BUILD_PATH_VAR="$V2_COMPILER_BUILD_PATH_VAR"
  local COMPILER_BUILD_PATH="${!COMPILER_BUILD_PATH_VAR}"
```

Reply:
but this variable has assigned a value of V1_COMPILER_BUILD_PATH_VAR, right? so the echo will print out the path and not the variable name (which I suppose you intend to do?)

Reply:
`COMPILER_BUILD_PATH_VAR` variable is assigned the name of either `V1_COMPILER_BUILD_PATH_VAR` or `V2_COMPILER_BUILD_PATH_VAR`, not value.

And it is intended to print the name, not path, otherwise it wouldn't give any info on what variable to set.

Reply:
ahh, I see now. You have a variable to hold a different variable name. I don't see ny reason why it should be done like this to be honest, it looks to me like a weird workaround. Why not just check for V1_COMPILER_BUILD_PATH or V2_COMPILER_BUILD_PATH existence, and use it as COMPILER_BUILD_PATH_VAR? the code would be less complicated, possibly even shorter